openstack: add terraform example (#13)
flatcar/flatcar-terraform: openstack/README.md

1. Clone the repo.
2. Add credentials in a `terraform.tfvars` file, expected credentials name can be found in `provider.tf`
3. Run
   ```shell
   terraform init
   ```
4. Edit [`server1.yaml`][server-1] and provide your own custom provisioning configuration in [container linux config][container-linux-config] syntax.
5. Plan and apply.
   This will also auto-generate an SSH key pair to log in to the server after provisioning.
   The public key of that key pair will be registered with OpenStack. <br />
   Invoke Terraform:
   ```shell
   terraform plan
   terraform apply
   ```

Terraform will print server information (name, ipv4 and v6) after deployment concluded.
The deployment will create an SSH key pair in `.ssh/`.

After provisioning concluded the private key of the key pair generated during provisioning can be used to connect to node. As mentioned in the disclaimer, the tested setup relies on DevStack so we need to proxy jump on the devstack instance before reaching the deployed instance:
```shell
ssh -J user@[DEVSTACK-IP] -i ./.ssh/provisioning_private_key.pem -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null core@[SERVER-IP]
```

_NOTE_:

> * Server IP can be find at any moment after deployment by running `terraform output`

Reviewer: ```suggestion
* Server IP address can be found at any moment after deployment by running `terraform output`
```

* If you update server configuration(s) in `server-configs` and re-run `terraform apply`, the instance will be **replaced**.
Consider adding [`create_before_destroy`](https://www.terraform.io/docs/configuration/meta-arguments/lifecycle.html#syntax-and-arguments) to the `openstack_compute_instance_v2` resource in [`compute.tf`](compute.tf) to avoid services becoming unavailable during reprovisioning.

> * SSH key is passed twice: in the container Linux configuration and in the `key_pair` argument of the instance - while the SSH keys in the container Linux configuration are added by Ignition during the first boot of the instance, `key_pair` are processed by `coreos-metadata-sshkeys@.service` (a.k.a [`afterburn`][afterburn]) from the OpenStack [metadata service][metadata-service]. It means that you can install SSH key without Ignition.

Reviewer: ```suggestion
* SSH key is passed twice: in the Butane configuration and in the `key_pair` argument of the instance - while the SSH keys in the Butane configuration are added by Ignition during the first boot of the instance, `key_pair` are processed by `coreos-metadata-sshkeys@.service` (a.k.a [`afterburn`][afterburn]) from the OpenStack [metadata service][metadata-service]. It means that you can install SSH keys without Ignition and change them as wanted.
```
